kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java

```diff
@@ -126,15 +126,25 @@ public ReplicaSet scale(int count, boolean wait) {
   /**
    * Lets wait until there are enough Ready pods
    */
-  private void waitUntilScaled() {
+  private void waitUntilScaled(final int count) {
     final CountDownLatch countDownLatch = new CountDownLatch(1);
 
     final AtomicReference<ReplicaSet> atomicRC = new AtomicReference<>();
 
     final Runnable rcPoller = new Runnable() {
       public void run() {
         try {
-          ReplicaSet rc = getMandatory();
+          ReplicaSet rc = get();
+          //If the rs is gone, we shouldn't wait.
+          if (rc == null) {
+            if (count == 0) {
+              countDownLatch.countDown();
+              return;
+            } else {
+              return;
```

Comment:
this is separate to passing the exception - by returning and not counting down the latch we'll block the caller thread forever right (exception or no exception)?